onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h

```diff
@@ -52,9 +52,15 @@ class NodeGroupSelector {
 // Single DQ -> node that does not change data -> Q.
 // Zero point and scale are constant scalars and must match
 class DropQDQNodeGroupSelector : public NodeGroupSelector {
+ public:
+  explicit DropQDQNodeGroupSelector(bool int16_uint16_allowed = true) : int16_uint16_allowed_(int16_uint16_allowed) {}
+
+ private:
   bool Check(const GraphViewer& graph_viewer, const Node& node,
              const std::vector<const Node*>& dq_nodes,
              const std::vector<const Node*>& q_nodes) const override;
+
+  bool int16_uint16_allowed_;
```

Comment:
I implemented something similar to your suggestion in my local branch but decided to hold it back. I am now not so sure that adding various `allow_intX_` bools to the base NodeGroupSelector class is appropriate for the following reasons:

1. The `allow_16bits_` bools were added only because we needed to disable most 16bit QDQ fusions (e.g., QLinear*). As we add 16bit support to QLinear ops in the future, these flags will probably be removed. There is currently no need to disable 16bit support in other contexts.

2. Having these bools in the base class creates a false sense that these bools apply equally to all inputs and outputs. However, this is not always the case. For example, in the ConvNodeGroupSelector the `allow_int8_` bool only applies to the first input (not the weight). However, the `allow_int16_` bool would confusingly apply to all inputs and outputs. I found myself having to write comments explaining why some of these inconsistencies.

By leaving the bools in the derived classes, it is more obvious (at least to me) that they have a context-specific meaning. For example, in conv the `allow_int8_` bool only refers to the input, not the weight.

I pushed this experiment into a separate branch in case you wanted to take a look: https://github.com/microsoft/onnxruntime/compare/adrianl/contrib-qdq16bit...adrianl/contrib-qdq16bit-experiment1